From 56031e3eb560d2be8ed44a6af40a1c0a8a43d51c Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <nathan@kernel.org>
Date: Thu, 4 Nov 2021 11:40:45 -0700
Subject: [PATCH] HACK: arch/arm: Assume __builtin_thread_pointer() is broken
 with LLVM 13.0.1 for now

This patch is only for our continuous integration.

Ard's fix in LLVM [1] has not been applied to the release/13.x branch
yet [2], even though it should given Peter Smith's approval, which
breaks our CI [3]. Once it lands in the release/13.x, it might take some
time for Debian's LLVM 13 to get the fix. To avoid breaking the build
while this whole process happens, apply this patch.

[1]: https://github.com/llvm/llvm-project/commit/d7e089f2d6a5cd5f283a90ab29241d20d4fc3ed1
[2]: https://bugs.llvm.org/show_bug.cgi?id=52329
[3]: https://github.com/ClangBuiltLinux/continuous-integration2/runs/4105274174?check_suite_focus=true

Signed-off-by: Nathan Chancellor <nathan@kernel.org>
---
 arch/arm/include/asm/current.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/include/asm/current.h b/arch/arm/include/asm/current.h
index 6bf0aad672c3..669c76ac6455 100644
--- a/arch/arm/include/asm/current.h
+++ b/arch/arm/include/asm/current.h
@@ -28,7 +28,7 @@ static inline struct task_struct *get_current(void)
 
 #if __has_builtin(__builtin_thread_pointer) && \
     !(defined(CONFIG_THUMB2_KERNEL) && \
-      defined(CONFIG_CC_IS_CLANG) && CONFIG_CLANG_VERSION < 130001)
+      defined(CONFIG_CC_IS_CLANG) && CONFIG_CLANG_VERSION < 140000)
 	/*
 	 * Use the __builtin helper when available - this results in better
 	 * code, especially when using GCC in combination with the per-task
-- 
2.34.0.rc0

